src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java

```diff
@@ -0,0 +1,419 @@
+/*
+ * Copyright © 2016-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.batch.mainframe.reader;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.common.ReferencePluginConfig;
+import net.sf.JRecord.Common.RecordException;
+import net.sf.JRecord.External.Def.ExternalField;
+import net.sf.JRecord.External.ExternalRecord;
+import org.apache.commons.io.IOUtils;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Mainframe Reader plugin configuration.
+ */
+public class MainframeSourceConfig extends ReferencePluginConfig {
+
+  public static final String REPLACEMENTS = "replacements";
+  public static final String BINARY_FILE_PATH = "binaryFilePath";
+  public static final String COPYBOOK_CONTENTS = "copybookContents";
+  public static final String DROP = "drop";
+  public static final String KEEP = "keep";
+  public static final String MAX_SPLIT_SIZE = "maxSplitSize";
+  public static final String CODEPAGE = "codepage";
+  public static final String CHARSET = "charset";
+
+  public static final long DEFAULT_MAX_SPLIT_SIZE_IN_MB = 1;
+  public static final long CONVERT_TO_BYTES = 1024 * 1024;
+  public static final Pattern RTRIM = Pattern.compile("\\s+$");
```

Comment:
Can we rename it to something more clear.

Reply:
Done.